src/apps/chatter/hooks/useWebSockets.js

```diff
@@ -0,0 +1,92 @@
+import url from '/util/url'
+import {
+  useRef
+} from 'react'
+
+export function useWebSockets (roomId) {
+  const socketUrl = url('api_ws.chatSocket', {
+    args: {
+      id: roomId.toString()
+    }
+  })
+  const callbacks = useRef({})
+  const socketRef = useRef()
+
+  const connect = () => {
```

Comment:
You will likely want to use `useEffect` inside your custom hook to bind to the websocket on mount. You will also want to think about if you want the websocket connection to be closed when the parent/calling component unmounts?

And, should this hook allow many websocket connections or only ever reference one open connection?